exercises/affine-cipher/src/test/java/AffineCipherTest.java

```diff
@@ -0,0 +1,136 @@
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import static org.junit.Assert.assertEquals;
+
+public class AffineCipherTest {
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    @Test
+    public void testEncodeYes() {
+        AffineCipher affineCipher = new AffineCipher();
+
+        assertEquals("xbt", affineCipher.encode("yes", 5, 7)
+        );
+    }
+
+    @Ignore("Remove to run test.")
+    @Test
+    public void testEncodeNo() {
+        AffineCipher affineCipher = new AffineCipher();
+        assertEquals("fu", affineCipher.encode("no", 15, 18));
+    }
+
+
+    @Ignore("Remove to run test.")
+    @Test
+    public void testEncodeOMG() {
+        AffineCipher affineCipher = new AffineCipher();
+        assertEquals("lvz", affineCipher.encode("OMG", 21, 3));
+    }
+
+    @Ignore("Remove to run test.")
+    @Test
+    public void testEncodeO_M_G() {
+        AffineCipher affineCipher = new AffineCipher();
+        assertEquals("hjp", affineCipher.encode("O M G", 25, 47));
+    }
+
+    @Ignore("Remove to run test.")
+    @Test
+    public void testEncodeMindBlowingly() {
+        AffineCipher affineCipher = new AffineCipher();
+        assertEquals("rzcwa gnxzc dgt", affineCipher.encode("mindblowingly", 11, 15));
+    }
+
+    @Ignore("Remove to run test.")
+    @Test
+    public void testEncodeNumbers() {
+        AffineCipher affineCipher = new AffineCipher();
+        assertEquals("jqgjc rw123 jqgjc rw",
+                affineCipher.encode("Testing,1 2 3, testing.", 3, 4));
+    }
+
+    @Ignore("Remove to run test.")
+    @Test
+    public void testEncodeDeepThought() {
+        AffineCipher affineCipher = new AffineCipher();
+        assertEquals("iynia fdqfb ifje",
+                affineCipher.encode("Truth is fiction.", 5, 17));
+    }
+
+    @Ignore("Remove to run test.")
+    @Test
+    public void testEncodeAllTheLetters() {
+        AffineCipher affineCipher = new AffineCipher();
+        assertEquals("swxtj npvyk lruol iejdc blaxk swxmh qzglf",
+                affineCipher.encode("The quick brown fox jumps over the lazy dog.", 17, 33));
+    }
+
+    @Ignore("Remove to run test.")
+    @Test
+    public void testEncodeThrowsMeaningfulException() throws IllegalArgumentException {
```

Comment:
Since `IllegalArgumentException` is an unchecked exception you don't need to include `throws` here 🙂